### PR TITLE
RFC: Disable zkApp Commands in case of Emergency

### PR DESCRIPTION
**Pull request by ejMina226**
Explain your changes:

Add the RFC for limiting zkApp commands fitting in a block.

### CODE REVIEW

File: rfcs/0056-disable-zkapp-commands.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+
+## Motivation
+Previously, there was a zkApp Softcap Limit that could be configured either on start-up of the mina node, or through an authenticated GraphQL endpoint. However, this is not safe enough as any block-producer running a node could just recompile the code and change the configuration, circumventing the zkApp command limit. Furthermore, the limit is *soft* in the sense that a mina node will still accept blocks which exceed the configured zkApp command limit. Therefore, another mechanism is required to ensure that any block producers who attempt to bypass the limit will not have their blocks accepted.
+
+## Detailed design
+Similar to the softcap case, the limit should be configurable at compile time, for example through a build profile. However, unlike in the softcap case the limit needs to be implemented at the block application level, rather than the block production level, as this change impacts non-BP mina nodes, as well. One candidate for the location is the `create_diff` function in the `staged_ledger.ml`. There is already a `Validate_and_apply_transactions` section in the function that could be coopted.
```
Comment (mrmr1993) [inline rfcs/0056-disable-zkapp-commands.md:8]: Can we put it in the runtime config? It has the same effect as being compile-time, but it makes it easier to maintain and simpler to release.

Comment (mrmr1993) [inline rfcs/0056-disable-zkapp-commands.md:8]: Nit: `s/coopted/co-opted`

**Review by mrmr1993 — COMMENTED** [review]
LGTM, but see comments

**Comment by ejMina226** [comment]
!ci-build-me

**Review by mrmr1993 — APPROVED** [review]

**Pull request merged**